app/queries/get_course_option_from_codes.rb

```diff
@@ -66,4 +57,26 @@ def initialize(
   def call
     course_option if valid?
   end
+
+  def self.get_unique_course_option(record, attr)
+    course_option_attrs = { study_mode: record.study_mode }
+
+    course_option_attrs[:site] = record.site if record.site
+
+    possible_course_options = record.course.course_options.where(course_option_attrs)
+
+    if possible_course_options.count == 1
+      record.course_option ||= possible_course_options.first
+    else
+      record.errors.add(attr, build_course_option_error_message(record, possible_course_options.count))
```

Comment:
We now handle the "ambiguous" and "not found" cases together, which has slightly changed the behaviour when providing a site. Previously we were doing a find_by, which does not care if the found record is the only one. Now we ensure the found option is unique. This shouldn't have any impact unless we have some bad data somewhere